### DISCUSSION TITLE
Pocketbase Framework "pb_hooks", new Date().toISOString() is subtracting for timezone

### DISCUSSION THREAD

**Discussion by jonshipman** [pocketbase/pocketbase#4244] [Q&A]
To test

test.pb.js
```
$app.rootCmd.addCommand(
	new Command({
		use: 'test',
		run(_, args) {
			const time = args[0] || '1970-01-01';

			const dateObj = new Date(time);
			dateObj.setHours(0, 0, 0, 0);

			console.log(dateObj.toISOString());
		}
	})
);
```

Running ./pocketbase test will output "1969-12-31T06:00:00.000Z" (I'm in CDT, YMMV). If I remove the dateObj.setHours I get the correct hours (for just dates) but passing in the time will resurface the issue, .e.g. `./pocketbase test 2024-01-01` will be different from `./pocketbase test "2024-01-01 00:00:00"`

```
PS> .\db\pocketbase test "2024-01-01 00:00:00"
2024-01-01T06:00:00.000Z
PS> .\db\pocketbase test "2024-01-01"         
2024-01-01T00:00:00.000Z
```

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I follow, but testing locally the below code all returns the same date for me:

```js
// pb_hooks/main.pb.js

const date1 = new Date("2024-01-01");
date1.setHours(0, 0, 0, 0);
console.log(date1.toISOString()); // 2023-12-31T22:00:00.000Z

const date2 = new Date("2024-01-01 00:00:00");
date2.setHours(0, 0, 0, 0);
console.log(date2.toISOString()); // 2023-12-31T22:00:00.000Z

const date3 = new Date("2024-01-01 00:00:00");
console.log(date3.toISOString()); // 2023-12-31T22:00:00.000Z
```
_Note also the `Date` caveat mentioned in the goja repo - https://github.com/dop251/goja?tab=readme-ov-file#date._

_edit: updated with a 3rd example without `setHours`._

**Reply by jonshipman** [reply]
Is your computer set to GMT/UTC?

**Reply by ganigeorgiev** [reply]
No, EET (confirmed by running `date`).

I'm testing on Linux (Fedora) so it could be something platform specific, I'm not really sure (_I haven't checked the goja date parser but if it relies on `time.LoadLocation` this may fail on Windows as in the past we had issues with that in https://github.com/pocketbase/pocketbase/issues/45#issuecomment-1179763680_).

Could you try to paste the above code snippet in the browser console to check if it will return different result?

**Reply by ganigeorgiev** [reply]
I don't have a Winows VM setup at the moment to test it but if the culprit ended up being the local timezone detection, then you can try:
- always specify a timezone part to your dates to avoid the local fallback
- if the above is not an option, you can try to manually download the tzdata archive (for example from the IANA website) and define the `ZONEINFO` env variable pointing to the unzipped directory (_alternatively, if you are building PocketBase manually you can also try adding the build tag `-tags timetzdata` but note that will increase the executable size with half a MB or so_).
